### DISCUSSION TITLE
Implementing multiplayer new design

### DISCUSSION THREAD

**Discussion by GSculerlor** [ppy/osu#17746] [Q&A]
hi! i'm looking into implementing the multiplayer screen new design based on this [WIP figma design](https://www.figma.com/file/XJCRjfvTlVIwZBWoAz5kB2/Client%2FPlaylist-Multiplayer?node-id=0%3A1) and have very early WIP changes on [my side](https://github.com/GSculerlor/osu/tree/room-search-textbox). have some question before going any further.

1. about the UI size, is there any conversion from Figma size to codebase size? seems like not everything that is already in `master` is 1:1 with the figma size
2. what's the expected design when [this dropdown](https://www.figma.com/file/XJCRjfvTlVIwZBWoAz5kB2/Client%2FPlaylist-Multiplayer?node-id=1025%3A5766) is expanded? is it going to be like what we already have or do they also have the shear effect applied?

**Comment by smoogipoo** [comment]
1: @bdach Able to provide some insight here based on your recent work with the mod select screen?
2: @arflyte

**Reply by bdach** [reply]
Dimensions in figma do not map directly to in game dimensions, if you want pixel for pixel accuracy with the end result.

For pixel-for-pixel accuracy at 1920x1080 I believe you need to downsize everything *but* font sizes by a factor of 1.4 times. Number is chosen based on the size of the internal draw size preserving container (which is 1366x768; 1920 / 1366 and 1080 / 768 both come up to about 1.4x).

**Reply by frenzibyte** [reply]
I'll also need this for my beatmap info overlay rework, so I'm going to ask here: ~~I'm slightly confused on one point, you mention that the internal size of the draw size preserving container is 1366x768, but isn't it 1024x768 according to `TargetDrawSize`?~~

Actually, I guess it is 1024x768 but adjusted to 1366x768 to match the aspect ratio of 1920x1080.

**Reply by bdach** [reply]
It's 1366x768 when examined in the draw visualizer at 1080p.

**Comment by GSculerlor** [comment]
bumping up this discussion for additional questions (more like bumping up my unanswered question)

currently, I have this on my local but I have several questions:
1. what's the expanded (menu) design look like? (bumping up my question on OP)
2. does the label have a fixed width or depend on the text's width? (or both?)
3. does the menu have the same width as the header (label+selected menu area) or have the same size as the selected menu area *only*? 
4. what's the hover state look like? 

attached screenshot of what's I currently have, the menu is from the currently implemented code
![image](https://user-images.githubusercontent.com/6506864/168336072-acc74716-6d87-4093-ae6a-23702762cb16.png)
![image](https://user-images.githubusercontent.com/6506864/168339707-d943f352-4099-4dea-86a8-b6ff3524c4b6.png)

also, should I convert it to issue to track the progress or just leave it as is?

**Reply by peppy** [reply]
@arflyte ping attempt 2

**Reply by arflyte** [reply]
I'll add the menu design into the figma file. Will ping once it's there.
